gdk4/src/rectangle.rs

```diff
@@ -0,0 +1,190 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use cairo::RectangleInt;
+use glib::ffi::gconstpointer;
+use glib::translate::*;
+use std::convert::{AsRef, From};
+use std::mem;
+
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
+#[repr(C)]
+pub struct Rectangle {
+    pub x: i32,
+    pub y: i32,
+    pub width: i32,
+    pub height: i32,
+}
+
+impl Rectangle {
+    #[doc(alias = "gdk_rectangle_contains_point")]
+    pub fn contains_point(&self, x: i32, y: i32) -> bool {
+        unsafe {
+            from_glib(ffi::gdk_rectangle_contains_point(
+                self.to_glib_none().0,
+                x,
+                y,
+            ))
+        }
+    }
+
+    #[doc(alias = "gdk_rectangle_equal")]
+    pub fn equal(&self, rect2: &Rectangle) -> bool {
```

Comment:
There's a reason why I didn't mention equal :) That's equivalent to the `#[derive(PartialEq)]` you already have, and having a separate `equal()` function (in addition to the `eq()` from `PartialEq`) is just confusing